src/main/java/slimeknights/tconstruct/smeltery/recipe/ICastingInventory.java

```diff
@@ -2,11 +2,21 @@
 
 import net.minecraft.fluid.Fluid;
 import slimeknights.tconstruct.library.recipe.inventory.ISingleItemInventory;
+import slimeknights.tconstruct.library.recipe.material.MaterialCastingRecipe;
+
+import javax.annotation.Nullable;
 
 public interface ICastingInventory extends ISingleItemInventory {
   /**
    * Gets the contained fluid in this inventory
    * @return  Contained fluid
    */
   Fluid getFluid();
+
+  /**
+   * Gets the material recipe based on the current slot contents
+   * @return  Material recipe
+   */
+  @Nullable
+  MaterialCastingRecipe getMaterial();
```

Comment:
Not sure how I feel about adding this to the casting inventory. We needed it in the part builder as materials are a primary function of the block, while this only uses them for one specialized recipe.

How much work would it be to find the material within the `PartCastingRecipe`? I guess we may have to do it twice then, once for input, once for output.